src/docker-osx-dev

```diff
@@ -63,6 +63,7 @@ DOCKER_HOST_USER=""
 DOCKER_HOST_SSH_URL=""
 DOCKER_HOST_SSH_KEY=""
 DOCKER_HOST_SSH_COMMAND=""
+REMOVE_SHARED_FOLDERS=""
```

Comment:
To keep the code a bit easier to understand and maintain, I try to reduce the use of global variables to just types that bash does not handle well, such as lists or maps. Any chance you could change this flag to a local variable and pass it explicitly through to the  `check_for_shared_folders` function?